### PR TITLE
Add option to enable path style access for hive s3

### PR DESCRIPTION
**Pull request by darklore**
Dear team,

The current hive connector have a property to set customized S3 endpoint for S3-compatible storage.
But, the  storage must have virtual host style access. (Example: http://bucketname.s3.amazonaws.com/path/to/object )
http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html

For these storage, this PR adds a property that enables path style access of s3 client in hive connector.
(Example: http://s3.amazonaws.com/bucketname/path/to/object )

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -203,6 +203,10 @@ Property Name                                Description
 ``hive.s3.signer-type``                      Specify a different signer type for S3-compatible storage.
                                              Example: ``S3SignerType`` for v2 signer type
 
+``hive.s3.path-style-access``                Use path style for S3 access. This is for S3-compatible storage
+                                             that doesn't support virtual host style access
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/connector/hive.rst]: this is called `virtual-hosted-style` in most of the s3 docs including the javadoc.

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -203,6 +203,10 @@ Property Name                                Description
 ``hive.s3.signer-type``                      Specify a different signer type for S3-compatible storage.
                                              Example: ``S3SignerType`` for v2 signer type
 
+``hive.s3.path-style-access``                Use path style for S3 access. This is for S3-compatible storage
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/connector/hive.rst]: Similar to the javadoc of `S3ClientOptions::setPathStyleAccess`, how about we say `Use path-style access for all requests to the S3-compatible storage.`?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java [hunk 3]
```diff
@@ -122,6 +124,20 @@ public void testInstanceCredentialsDisabled()
         }
     }
 
+    @Test
+    public void testPathStyleAccess()
+            throws Exception
+    {
+        Configuration config = new Configuration();
+        config.set(PrestoS3FileSystem.S3_PATH_STYLE_ACCESS, "true");
```
Comment (nezihyigitbasi) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java]: import static `PrestoS3FileSystem.S3_PATH_STYLE_ACCESS`.

Comment (nezihyigitbasi) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java]: Also use `config.setBoolean(S3_PATH_STYLE_ACCESS, true);`

**Comment by darklore** [comment]
Thank you for reviewing.

I updated the patch according to review comments.

**Comment by nezihyigitbasi** [comment]
Can you squash the commits into one? Please use the guidelines [here](https://chris.beams.io/posts/git-commit/) when writing your commit message.

**Comment by darklore** [comment]
I have squashed them.

**Comment by nezihyigitbasi** [comment]
Looks good, thanks! I will merge this after the current release goes out.

**Comment by nezihyigitbasi** [comment]
@darklore can you please rebase this? There are some conflicts to be resolved.

**Comment by darklore** [comment]
Thank you for the notice.
I have rebased.

**Comment by nezihyigitbasi** [comment]
Thanks @darklore!

**Pull request merged**